Source/Core/DolphinQt2/Debugger/MemoryWidget.cpp

```diff
@@ -0,0 +1,565 @@
+  connect(m_find_ascii, &QRadioButton::toggled, this, &MemoryWidget::ValidateSearchValue);
+  connect(m_find_hex, &QRadioButton::toggled, this, &MemoryWidget::ValidateSearchValue);
+
+  connect(m_set_value, &QPushButton::pressed, this, &MemoryWidget::OnSetValue);
+
+  connect(m_dump_mram, &QPushButton::pressed, this, &MemoryWidget::OnDumpMRAM);
+  connect(m_dump_exram, &QPushButton::pressed, this, &MemoryWidget::OnDumpExRAM);
+  connect(m_dump_fake_vmem, &QPushButton::pressed, this, &MemoryWidget::OnDumpFakeVMEM);
+
+  connect(m_find_next, &QPushButton::pressed, this, &MemoryWidget::OnFindNextValue);
+  connect(m_find_previous, &QPushButton::pressed, this, &MemoryWidget::OnFindPreviousValue);
+
+  for (auto* radio : {m_type_u8, m_type_u16, m_type_u32, m_type_ascii, m_type_float})
+    connect(radio, &QRadioButton::toggled, this, &MemoryWidget::OnTypeChanged);
+
+  for (auto* radio : {m_bp_read_write, m_bp_read_only, m_bp_write_only})
+    connect(radio, &QRadioButton::toggled, this, &MemoryWidget::OnBPTypeChanged);
+
+  connect(m_bp_log_check, &QCheckBox::toggled, this, &MemoryWidget::OnBPLogChanged);
+  connect(m_memory_view, &MemoryViewWidget::BreakpointsChanged, this,
+          &MemoryWidget::BreakpointsChanged);
+}
+
+void MemoryWidget::closeEvent(QCloseEvent*)
+{
+  Settings::Instance().SetMemoryVisible(false);
+}
+
+void MemoryWidget::Update()
+{
+  m_memory_view->Update();
+}
+
+void MemoryWidget::LoadSettings()
+{
+  QSettings& settings = Settings::GetQSettings();
+
+  const bool search_ascii =
+      settings.value(QStringLiteral("memorywidget/searchascii"), true).toBool();
+  const bool search_hex = settings.value(QStringLiteral("memorywidget/searchhex"), false).toBool();
+
+  m_find_ascii->setChecked(search_ascii);
+  m_find_hex->setChecked(search_hex);
+
+  const bool type_u8 = settings.value(QStringLiteral("memorywidget/typeu8"), true).toBool();
+  const bool type_u16 = settings.value(QStringLiteral("memorywidget/typeu16"), false).toBool();
+  const bool type_u32 = settings.value(QStringLiteral("memorywidget/typeu32"), false).toBool();
+  const bool type_float = settings.value(QStringLiteral("memorywidget/typefloat"), false).toBool();
+  const bool type_ascii = settings.value(QStringLiteral("memorywidget/typeascii"), false).toBool();
+
+  m_type_u8->setChecked(type_u8);
+  m_type_u16->setChecked(type_u16);
+  m_type_u32->setChecked(type_u32);
+  m_type_float->setChecked(type_float);
+  m_type_ascii->setChecked(type_ascii);
+
+  bool bp_rw = settings.value(QStringLiteral("memorywidget/bpreadwrite"), true).toBool();
+  bool bp_r = settings.value(QStringLiteral("memorywidget/bpread"), false).toBool();
+  bool bp_w = settings.value(QStringLiteral("memorywidget/bpwrite"), false).toBool();
+  bool bp_log = settings.value(QStringLiteral("memorywidget/bplog"), true).toBool();
+
+  m_bp_read_write->setChecked(bp_rw);
+  m_bp_read_only->setChecked(bp_r);
+  m_bp_write_only->setChecked(bp_w);
+  m_bp_log_check->setChecked(bp_log);
+}
```

Comment:
Settings are not applied to MemoryViewWidget when relaunch dolphin.

Reply:
You do mean ``MemoryWidget`` right? Anyway it does load and save properly here.

Reply:
Data Type always becomes U8 after relaunch dolphin regardless of settings.
![snapcrab_noname_2018-3-29_13-33-18_no-00](https://user-images.githubusercontent.com/21278142/38070586-ac0b40fc-3357-11e8-9980-2f0f25a5820b.png)

Reply:
Misunderstood you there, sorry! Anyway fixed now.